services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -0,0 +1,970 @@
+
+    config_names_content = response["content"]
+    if "config_names" not in config_names_content:
+        raise PreviousStepFormatError("Previous step did not return the expected content: 'config_names'.")
+
+    if not isinstance(config_names_content["config_names"], list):
+        raise PreviousStepFormatError(
+            "Previous step did not return the expected content.",
+            TypeError(f"config_names should be a list, but got {type(config_names_content['config_names'])}"),
+        )
+
+    config_names = {config_name_item["config"] for config_name_item in config_names_content["config_names"]}
+    if config not in config_names:
+        raise ConfigNamesError(f"{config=} does not exist in {dataset=}")
+
+    hf_api = HfApi(endpoint=hf_endpoint, token=hf_token)
+    committer_hf_api = HfApi(endpoint=hf_endpoint, token=committer_hf_token)
+
+    # prepare the parquet files locally
+    local_parquet_files: List[ParquetFile] = []
+    download_config = DownloadConfig(delete_extracted=True)
+    try:
+        builder = load_dataset_builder(
+            path=dataset,
+            name=config,
+            revision=source_revision,
+            use_auth_token=hf_token,
+            download_config=download_config,
+        )
+    except _EmptyDatasetError as err:
+        raise EmptyDatasetError(f"{dataset=} is empty.", cause=err) from err
+    raise_if_too_big_from_external_data_files(
+        builder=builder,
+        max_dataset_size=max_dataset_size,
+        max_external_data_files=max_external_data_files,
+        hf_token=hf_token,
+    )
+    builder.download_and_prepare(file_format="parquet")  # the parquet files are stored in the cache dir
+    dataset_info = asdict(builder.info)
+    local_parquet_files.extend(
+        ParquetFile(local_file=local_file, local_dir=builder.cache_dir, config=config)
+        for local_file in glob.glob(f"{builder.cache_dir}**/*.parquet")
+    )
+
+    # create the target revision if it does not exist yet
+    try:
+        refs = hf_api.list_repo_refs(repo_id=dataset, repo_type=DATASET_TYPE)
+        if all(ref.ref != target_revision for ref in refs.converts):
+            committer_hf_api.create_branch(
+                repo_id=dataset, branch=target_revision, repo_type=DATASET_TYPE, revision=source_revision
+            )
+    except RepositoryNotFoundError as err:
+        raise DatasetNotFoundError("The dataset does not exist on the Hub.") from err
+
+    target_dataset_info = hf_api.dataset_info(repo_id=dataset, revision=target_revision, files_metadata=False)
+    # - check if there are files for configs that do not exist anymore and delete them
+    # - get configs that exist in repo
+    repo_parquet_files = {f.rfilename for f in target_dataset_info.siblings if f.rfilename.endswith(".parquet")}
+    repo_config_names = {f.split("/")[0] for f in repo_parquet_files}
+    delete_operations: List[CommitOperation] = []
+    legacy_configs = repo_config_names - config_names
```

Comment:
here we rely on the fact that the directory names are exactly the config names. Is is always the case? Or can they be weird characters in a config name that needs to be normalized (and is normalized by the datasets library)? In which case we would have to compare the normalized configs (ie. the directory names)